frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -965,34 +968,45 @@ pub mod pallet {
             GeneralEraInfo::<T>::insert(era, era_info);
         }
 
-        /// This helper returns `ContractStakeInfo` for given era if possible or latest stored data
-        /// or finally default value if storage have no data for it.
+        /// Used to copy all `ContractStakeInfo` from the ending era over to the next era.
+        /// This is the most primitive solution since it scales with number of dApps.
+        /// It is possible to provide a hybrid solution which allows laziness but also prevents
+        /// a situation where we don't have access to the required data.
+        fn advance_staking_info(current_era: EraIndex) -> u64 {
```

Comment:
May be `rotate_staking_info`?

Reply:
I can rename it to that, sure, but we don't just _rotate_, we also do copy.